test/functional/aws_test.go

```diff
@@ -0,0 +1,85 @@
+package functional
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/elasticache/elasticacheiface"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/docker/docker/client"
+	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	"github.com/integr8ly/integreatly-operator/test/common"
+	"testing"
+)
+
+//TODO update when Fuse Online is ready as per https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases/blob/master/tests/high-availability/f01-verify-all-aws-resources-are-created-in-aws.md
+const (
+	tagKeyClusterId            = "integreatly.org/clusterID"
+	RDSSuffix                  = "postgres-integreatly"
+	UPSResourceName            = "ups-" + RDSSuffix
+	CRResourceName             = "codeready-" + RDSSuffix
+	ClusterSSOResourceName     = "rhsso-" + RDSSuffix
+	ApplicationSSOResourceName = "rhssouser-" + RDSSuffix
+	ThreeScaleNameElement      = "rhmioperatorthre"
+	S3Backup                   = "rhmioperatorback"
+	loggingKeyDatabase         = "database-id"
+)
+
+func TestAWS(t *testing.T) {
+	//TODO
+}
+
+type awsClients struct {
+	rdsClient         rdsiface.RDSAPI
+	elasticacheClient elasticacheiface.ElastiCacheAPI
+	s3Client          s3iface.S3API
+	taggingClient     resourcegroupstaggingapi.ResourceGroupsTaggingAPI
+}
+
+func getRDSResources(c *awsClients, t *testing.T, tags map[string]string) ([]*rds.DBInstance, error) {
+	var clusterID = resources.GetClusterID
+	var clusterIDString = fmt.Sprintf("%d", clusterID)
+
+	clusterDescribeInput := &rds.DescribeDBInstancesInput{}
+	clusterDescribeOutput, err := c.rdsClient.DescribeDBInstances(clusterDescribeInput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe database clusters", err)
+	}
+	var RDSDBInstances []*rds.DBInstance
+	for _, dbInstance := range clusterDescribeOutput.DBInstances {
+		t.Log(loggingKeyDatabase, aws.StringValue(dbInstance.DBInstanceIdentifier))
+		t.Log("checking tags database cluster")
+		tagListInput := &rds.ListTagsForResourceInput{
+			ResourceName: dbInstance.DBInstanceArn,
+		}
+		tagListOutput, err := c.rdsClient.ListTagsForResource(tagListInput)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list tags for database cluster", err)
+		}
+		t.Logf("checking for cluster tag match (%s=%s) on database", tagKeyClusterId, clusterIDString)
+		if common.FindTag(tagKeyClusterId, clusterIDString, tagListOutput.TagList) == nil {
+			t.Logf("database did not contain cluster tag match (%s=%s)", tagKeyClusterId, clusterIDString)
+			continue
+		}
+		extraTagsMatch := true
+		for extraTagKey, extraTagVal := range tags {
```

Comment:
Do we need these extra tag checks as well? I would just be in favour of returning the entire list of found DB instances and then checking if their names/tags matched the ones we expected in the main function :thinking: